src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentSynchronization/RazorDidOpenTextDocumentEndpoint.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts;
+using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.DocumentSynchronization;
+
+internal class RazorDidOpenTextDocumentEndpoint : IVSDidOpenTextDocumentEndpoint
+{
+    public bool MutatesSolutionState => true;
+
+    private readonly ProjectSnapshotManagerDispatcher _projectSnapshotManagerDispatcher;
+    private readonly RazorProjectService _projectService;
+
+    public RazorDidOpenTextDocumentEndpoint(ProjectSnapshotManagerDispatcher projectSnapshotManagerDispatcher, RazorProjectService razorProjectService)
+    {
+        _projectSnapshotManagerDispatcher = projectSnapshotManagerDispatcher;
+        _projectService = razorProjectService;
+    }
+
+    public TextDocumentIdentifier GetTextDocumentIdentifier(DidOpenTextDocumentParamsBridge request)
+    {
+        var identifier = new TextDocumentIdentifier
+        {
+            Uri = request.TextDocument.Uri,
```

Comment:
It is not, it's a `TextDocumentItem`. I could add TextDocumentItem to the list of things handled by the RequestContextFactory, but I figured this was cleaner since if I remember right this is the only spot that uses TextDocumentItem.

Reply:
In the end since I added Uri for another comment you had I figured I could just return the Uri here.